mne/encoding/feature.py

```diff
@@ -108,50 +123,80 @@ def fit_transform(self, event_ixs, event_ids=None, event_dict=None):
 
 
 class DataSubsetter(object):
-    """Return a subset of data."""
+    """Use for returning a subset of data.
+
+    Parameters
+    ----------
+    ixs : array, shape (n_ixs,)
+        The indices to select from the data."""
     def __init__(self, ixs):
         self.ixs = ixs.astype(int)
 
     def fit(self, data, y=None):
+        """Pull a subset of data points.
+
+        Parameters
+        ----------
+        data : array, shape (n_features, n_samples)
+            The data to be subsetted. The data points specified by
+            `self.ixs` will be taken from columns of this array.
+        """
         if data.shape[-1] < self.ixs.max():
             raise ValueError('ixs exceed data shape')
         self.data = np.asarray(data)
 
     def transform(self, X=None, y=None):
+        """Return a subset of data points.
+
+        Returns
+        -------
+        data : array, shape (n_features, n_ixs)
+            A subset of columns from the input array.
+        """
         return self.data[..., self.ixs]
 
     def fit_transform(self, ixs, y=None):
         self.fit(ixs)
         return self.transform()
 
 
-def delay_timeseries(ts, sfreq, delays):
+def _delay_timeseries(ts, delays, sfreq=1.):
     """Return a time-lagged input timeseries.
 
     Parameters
     ----------
     ts: array, shape (n_feats, n_times)
         The timeseries to delay
+    delays: array of floats, shape (n_delays,)
+        The time (in seconds) of each delay for specifying
+        pre-defined delays. Negative means timepoints in the past,
+        positive means timepoints in the future.
     sfreq: int
         The sampling frequency of the series
-    delays: list of floats
-        The time (in seconds) of each delay. Negative means
-        timepoints in the past, positive means timepoints in
-        the future.
 
     Returns
     -------
     delayed: array, shape(n_feats * n_delays, n_times)
         The delayed matrix
     """
-    delayed = np.zeros([len(delays), ts.shape[-1]])
-    for ii, delay in enumerate(delays):
-        roll_amount = -1 * int(delay * sfreq)
+    n_feats, n_times = ts.shape
+    delays_ixs = -1 * (delays * sfreq).astype(int)
```

Comment:
I was using -1 because if we want negative numbers to mean delays in the past, then we need to shift the timeseries forward, which means we need a positive number to pass to `roll`. I think I'm just going to re-do this to avoid using roll and be more explicit.